SleepEngine/Engine/GameTemplate/Game.cpp

```diff
@@ -49,9 +98,11 @@ void Game::runFrame()
     EASY_FUNCTION(profiler::colors::Orange);
     m_clock.frameStart();
 
-    Base::update(m_clock.getDT());
-
-    m_renderer->render();
+    if (m_currentScene != ScenesContainer::iterator())
+    {
+        m_currentScene->second.first.update(m_clock.getDT());
+        m_renderer->render();
+    }
```

Comment:
it seems that default constructed iterator equals the cend() iter:
https://stackoverflow.com/questions/3395180/what-is-an-iterators-default-value
so i think it's more clear to compare it to cend(), as we always do

Reply:
How can it equal to cend() of an exact container if this iterator is fully independent of some exact container? I store this iterator as a member of my class and I'm sure it equals to default constructed value of this iterator)

Reply:
Oh, indeed its not true. But the thing you do is UB according to the second answer in my SO link

Reply:
Fixed